frontend/fleet/request.js

```diff
@@ -44,10 +44,15 @@ export default class Request {
   send() {
     const { endpoint, requestAttributes } = this;
 
-    return fetch(endpoint, requestAttributes).then((response) => {
-      return response.json().then((jsonResponse) => {
-        return Request.handleResponse(response, jsonResponse);
-      });
+    return fetch(endpoint, requestAttributes).then(async (response) => {
+      let jsonResponse;
+      try {
+        jsonResponse = await response.json();
+      } catch (error) {
+        console.log("Failed to parse response body as JSON: ", error);
+        throw response.statusText || error;
+      }
```

Comment:
I'm adding the try catch block around the JSON conversion only so that we can return the entire promise without catching any errors. If we catch the error on `Request.handleResponse`, error handling further downstream in the redux state fails to parse error responses properly.